### ISSUE TITLE
502 Bad Gateway for potentially big push

### ISSUE THREAD

**Issue by dzmitry-lahoda** [cachix/cachix#505]
We get
```
(retry #3) compressing using zstd and pushing /nix/store/w0ffipwcx6736pd6c8yr455civ8n93dy-composable-deps-2.10005.0 (6.01 GiB)
cachix: FailureResponse (Request {requestPath = (BaseUrl {baseUrlScheme = Https, baseUrlHost = "api.cachix.org", baseUrlPort = 443, baseUrlPath = ""},"/api/v1/cache/composable-community/nar"), requestQueryString = fromList [("compression",Just "zst")], requestBody = Just ((),application/octet-stream), requestAccept = fromList [], requestHeaders = fromList [("Authorization","<REDACTED>")], requestHttpVersion = HTTP/1.1, requestMethod = "POST"}) (Response {responseStatusCode = Status {statusCode = 502, statusMessage = "Bad Gateway"}, responseHeaders = fromList [("Server","nginx"),("Date","Wed, 08 Feb 2023 11:22:29 GMT"),("Content-Type","text/html; charset=utf-8"),("Content-Length","150"),("Connection","keep-alive")], responseHttpVersion = HTTP/1.1, responseBody = "<html>\r\n<head><title>502 Bad Gateway</title></head>\r\n<body>\r\n<center><h1>502 Bad Gateway</h1></center>\r\n<hr><center>nginx</center>\r\n</body>\r\n</html>\r\n"})
```
and  other run

```
compressing using zstd and pushing /nix/store/n9p5g5p527qc6rq926hk2ia3ckbgbdyx-run-benchmarks-once (784.00 B)
compressing using zstd and pushing /nix/store/w0ffipwcx6736pd6c8yr455civ8n93dy-composable-deps-2.10005.0 (6.01 GiB)
compressing using zstd and pushing /nix/store/k90wdfhxs42v27wx2ddzxqfwcladfk40-composable (968.14 MiB)
compressing using zstd and pushing /nix/store/ap04vpda15d83kgybm3wws22lvzl3614-run-benchmarks-once (784.00 B)
compressing using zstd and pushing /nix/store/vc4j97w1qf2244kz5l9dfy78h6s5pvji-dali-local-integration-tests-build-2.10005.0 (96.00 B)
compressing using zstd and pushing /nix/store/6vcy71s3q1f0148vm29mfvavkays4mx2-check-composable-benchmarks-ci (49.32 KiB)
compressing using zstd and pushing /nix/store/mwv17av2hhnpmlvh68dczii6ir9dh66v-check-picasso-benchmarks-ci (84.34 KiB)
compressing using zstd and pushing /nix/store/pryadxga96axnv0ngb30n06i8qy9pi6p-picasso-local-integration-tests-build-2.10005.0 (96.00 B)
compressing using zstd and pushing /nix/store/3c4indr762910nh50dllxd4rwks24x4f-check-dali-benchmarks-ci (159.30 KiB)
(retry #1) compressing using zstd and pushing /nix/store/k90wdfhxs42v27wx2ddzxqfwcladfk40-composable (968.14 MiB)
(retry #2) compressing using zstd and pushing /nix/store/k90wdfhxs42v27wx2ddzxqfwcladfk40-composable (968.14 MiB)
(retry #1) compressing using zstd and pushing /nix/store/w0ffipwcx6736pd6c8yr455civ8n93dy-composable-deps-2.10005.0 (6.01 GiB)
(retry #1) compressing using zstd and pushing /nix/store/w0ffipwcx6736pd6c8yr455civ8n93dy-composable-deps-2.10005.0 (6.01 GiB)
(retry #2) compressing using zstd and pushing /nix/store/w0ffipwcx6736pd6c8yr455civ8n93dy-composable-deps-2.10005.0 (6.01 GiB)
(retry #1) compressing using zstd and pushing /nix/store/lxbm27i3lqlw81qfr3bdjc4l70ghvx4n-composable-deps-2.10005.0 (6.48 GiB)
compressing using zstd and pushing /nix/store/m39b2n4k8g5d6350lm3zkqqlj30yvh4v-composable-build-2.10005.0 (96.00 B)
compressing using zstd and pushing /nix/store/m39b2n4k8g5d6350lm3zkqqlj30yvh4v-composable-build-2.10005.0 (96.00 B)
compressing using zstd and pushing /nix/store/hwrzzxcrk4a842s09nb8zs82m7f6xg5z-all-ci-packages (9.17 KiB)
(retry #3) compressing using zstd and pushing /nix/store/w0ffipwcx6736pd6c8yr455civ8n93dy-composable-deps-2.10005.0 (6.01 GiB)
(retry #1) compressing using zstd and pushing /nix/store/w0ffipwcx6736pd6c8yr455civ8n93dy-composable-deps-2.10005.0 (6.01 GiB)
(retry #4) compressing using zstd and pushing /nix/store/w0ffipwcx6736pd6c8yr455civ8n93dy-composable-deps-2.10005.0 (6.01 GiB)
(retry #2) compressing using zstd and pushing /nix/store/w0ffipwcx6736pd6c8yr455civ8n93dy-composable-deps-2.10005.0 (6.01 GiB)
(retry #2) compressing using zstd and pushing /nix/store/lxbm27i3lqlw81qfr3bdjc4l70ghvx4n-composable-deps-2.10005.0 (6.48 GiB)
(retry #3) compressing using zstd and pushing /nix/store/lxbm27i3lqlw81qfr3bdjc4l70ghvx4n-composable-deps-2.10005.0 (6.48 GiB)
(retry #4) compressing using zstd and pushing /nix/store/lxbm27i3lqlw81qfr3bdjc4l70ghvx4n-composable-deps-2.10005.0 (6.48 GiB)
(retry #5) compressing using zstd and pushing /nix/store/lxbm27i3lqlw81qfr3bdjc4l70ghvx4n-composable-deps-2.10005.0 (6.48 GiB)
(retry #5) compressing using zstd and pushing /nix/store/w0ffipwcx6736pd6c8yr455civ8n93dy-composable-deps-2.10005.0 (6.01 GiB)
(retry #3) compressing using zstd and pushing /nix/store/w0ffipwcx6736pd6c8yr455civ8n93dy-composable-deps-2.10005.0 (6.01 GiB)
cachix: FailureResponse (Request {requestPath = (BaseUrl {baseUrlScheme = Https, baseUrlHost = "api.cachix.org", baseUrlPort = 443, baseUrlPath = ""},"/api/v1/cache/composable-community/nar"), requestQueryString = fromList [("compression",Just "zst")], requestBody = Just ((),application/octet-stream), requestAccept = fromList [], requestHeaders = fromList [("Authorization","<REDACTED>")], requestHttpVersion = HTTP/1.1, requestMethod = "POST"}) (Response {responseStatusCode = Status {statusCode = 502, statusMessage = "Bad Gateway"}, responseHeaders = fromList [("Server","nginx"),("Date","Wed, 08 Feb 2023 11:22:29 GMT"),("Content-Type","text/html; charset=utf-8"),("Content-Length","150"),("Connection","keep-alive")], responseHttpVersion = HTTP/1.1, responseBody = "<html>\r\n<head><title>502 Bad Gateway</title></head>\r\n<body>\r\n<center><h1>502 Bad Gateway</h1></center>\r\n<hr><center>nginx</center>\r\n</body>\r\n</html>\r\n"})
error: writing to file: Broken pipe
error: writing to file: Broken pipe
error: writing to file: Broken pipe
will try execute 'cachix watch-exec --jobs 16 --compression-level 3 composable-community nix -- build .#all --no-http2 --no-update-lock-file  -L 2> >(tee --append  >&2)'
Error: Process completed with exit code 1.
```

Do we hit limit like https://github.com/cachix/cachix/issues/322 ? Or some rate limit?  Does cachix has some STATUS page to monitor  if something is down (like here https://github.com/cachix/cachix/issues/117 ) ?

**Comment by dzmitry-lahoda** [comment]
<img width="974" alt="image" src="https://user-images.githubusercontent.com/757125/217583831-a1c36f4a-293a-474a-b303-02402757d84a.png">

We are on Pro plan. Seems something is limited? Compute power may for decompression? Or somethin on Cloudflare side? May be size of upload? I guess it should work ok with big VM/docker images or big vendor folders. Rust can have 10GB vendors easy.

**Comment by domenkozar** [comment]
There are no limits, I'll look into it.

**Comment by domenkozar** [comment]
@dzmitry-lahoda do you have some timestamps for these failures?

@sandydoo is currently working on a big improvements for bigger files, but 6GB should be possible to upload.

**Comment by dzmitry-lahoda** [comment]
This one:

> cachix: FailureResponse (Request {requestPath = (BaseUrl {baseUrlScheme = Https, baseUrlHost = "api.cachix.org", baseUrlPort = 443, baseUrlPath = ""},"/api/v1/cache/composable-community/nar"), requestQueryString = fromList [("compression",Just "zst")], requestBody = Just ((),application/octet-stream), requestAccept = fromList [], requestHeaders = fromList [("Authorization","<REDACTED>")], requestHttpVersion = HTTP/1.1, requestMethod = "POST"}) (Response {responseStatusCode = Status {statusCode = 502, statusMessage = "Bad Gateway"}, responseHeaders = fromList [("Server","nginx"),("Date","Wed, 08 Feb 2023 11:22:29 GMT"),("Content-Type","text/html; charset=utf-8"),("Content-Length","150"),("Connection","keep-alive")], responseHttpVersion = HTTP/1.1, responseBody = "<html>\r\n<head><title>502 Bad Gateway</title></head>\r\n<body>\r\n<center><h1>502 Bad Gateway</h1></center>\r\n<hr><center>nginx</center>\r\n</body>\r\n</html>\r\n"})

> "Date","Wed, 08 Feb 2023 11:22:29 GMT")

**Comment by domenkozar** [comment]
We've released Cachix 1.3 that should fix these issues: https://blog.cachix.org/posts/2023-03-08-cachix-1-3-uploads-unleashed/
